taskcluster/utils.py

```diff
@@ -268,6 +270,45 @@ def isExpired(certificate):
   return expiry < int(time.time() * 1000 + 20 * 60 * 1000)
 
 
+def readCfgFile():
+  try:
+    with open(CONFIG_FILE) as cfgfile:
+      return json.load(cfgfile)['credentials']
+  except:
+    log.warn('Unable to read credentials file' + CONFIG_FILE)
+    raise
+
+
+def writeCfgFile(credentials):
+  data = {}
+  try:
+    try:
+      os.makedirs(os.path.dirname(CONFIG_FILE))
+    except:
+      pass
+    if os.path.exists(CONFIG_FILE):
+      with open(CONFIG_FILE) as cfgfile:
+        data = json.load(cfgfile)
+    data['credentials'] = credentials
+    with open(CONFIG_FILE, 'w') as cfgfile:
+      json.dump(data, cfgfile, indent=2)
+  except:
+    log.warn('Unable to write to config file ' + CONFIG_FILE)
+    raise
+
+
+def readCredentialsFromEnv():
+  c = {}
+  if 'TASKCLUSTER_CLIENT_ID' in os.environ and 'TASKCLUSTER_ACCESS_TOKEN' in os.environ:
+    c['clientId'] = os.environ['TASKCLUSTER_CLIENT_ID']
+    c['accessToken'] = os.environ['TASKCLUSTER_ACCESS_TOKEN']
+    if 'TASKCLUSTER_CERTIFICATE' in os.environ:
+        c['certificate'] = os.environ['TASKCLUSTER_CERTIFICATE']
+    return c
+  else:
+    return {}
+
+
 def authenticate(description=None):
```

Comment:
How about `configFile` as option defaulting to `CONFIG_FILE`, and have that `configFile = None` means we won't attempt to load from config file...

Option to load from env would also be nice, we can default it to `True`, as that is what most people will want.